frontend/components/main.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Meta from './meta'
+import Header from './header'
+import SuggestionAndSearch from './suggestion-and-search'
+import MarkupModal from './markup-modal'
+import Usage from './usage'
+import Footer from './footer'
+import { CategoryHeading, CategoryHeadings } from './category-headings'
+import {
+  categories,
+  findCategory,
+  services,
+  getDefinitionsForCategory,
+} from '../lib/service-definitions'
+import BadgeExamples from './badge-examples'
+import { baseUrl, longCache } from '../constants'
+import ServiceDefinitionSetHelper from '../lib/service-definitions/service-definition-set-helper'
+import groupBy from 'lodash.groupby'
+
+export default class Main extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isSearchInProgress: false,
+      isQueryTooShort: false,
+      searchResults: undefined,
+      selectedExample: undefined,
+    }
+
+    this.searchTimeout = 0
+
+    this.handleExampleSelected = this.handleExampleSelected.bind(this)
+    this.dismissMarkupModal = this.dismissMarkupModal.bind(this)
+    this.searchQueryChanged = this.searchQueryChanged.bind(this)
+  }
+
+  static propTypes = {
+    match: PropTypes.object.isRequired,
+  }
+
+  get category() {
+    return this.props.match.params.category
+  }
+
+  performSearch(query) {
+    const isQueryTooShort = query.length === 1
+
+    let searchResults
+    if (query.length >= 2) {
+      const flat = (searchResults = ServiceDefinitionSetHelper.create(services)
+        .notDeprecated()
+        .search(query)
+        .asNative())
+      searchResults = groupBy(flat, 'category')
+    }
+
+    this.setState({
+      isSearchInProgress: false,
+      isQueryTooShort,
+      searchResults,
+    })
+  }
+
+  searchQueryChanged(query) {
+    /*
+    Add a small delay before showing search results
+    so that we wait until the user has stipped typing
```

Comment:
```suggestion
    so that we wait until the user has stopped typing
```

this was probably originally my typo, but I just spotted it :)